src/snowflake/snowpark/mock/_functions.py

```diff
@@ -766,7 +766,11 @@ def mock_to_char(
             )
         func = partial(try_convert, lambda x: str(x), try_cast)
     else:
-        func = partial(try_convert, lambda x: str(x), try_cast)
+        func = partial(
```

Comment:
I prefer if we add an additional case for `BooleanType` where we use `str(x).lower()`
and an additional case for `VariantType` where we use `json.dumps(cls=CustomJsonEncoder)`.

Reply:
good idea, updated